### ISSUE TITLE
Detekt fails to catch a nullable string passed to a function that only takes string

### ISSUE THREAD

**Issue by rooday-doordash** [detekt/detekt#7707]
## Expected Behavior
If I try
```
nullableVar.let {
    functionThatTakesNonNullVar(it)
}
```
I should get an error

## Observed Behavior
I did not get an error. To be more specific, I did this:
```
SomeProtobufBuilder.apply {
    nullableString.let {
        this.stringProperty = StringValue.of(it)
    }
}
```
My code should have used `nullableString?.let`, but I missed this, and neither IntelliJ nor detekt caught this error, leading to NPEs later when the code was run.

## Steps to Reproduce

Try calling com.google.protobuf.StringValue.of on a nullable string and notice that detekt (even with type resolution) will not catch it.

## Context

This should be caught to prevent unsafe access/NPEs from happening.

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.8
* Gradle scan link (add `--scan` option when running the gradle task): unsure if I can share this
* Operating System and version: Mac OS X 14.7 aarch64
* Link to your project (if it's a public repository): private

**Comment by BraisGabin** [comment]
What is `StringValue.of(it)`? Does the project compile if you write `StringValue.of(null)`? If it compiles there's nothing on our side to fix that. The api allow null values so we don't report anything.

The only rule that could help you in your particular case is unnecessary let. Detekt would tell you that you could remove that let because it does nothing and that should warn you that something is off.

**Comment by rooday-doordash** [comment]
@BraisGabin you're right, `StringValue.of(null)` compiles and only during runtime we get a `java.lang.NullPointerException`. I guess I'll flag this with the protobuf repo then, perhaps they need to make an update?

**Comment by 3flex** [comment]
I'll close this then as it appears to be an issue in a different project.
